src/sql/workbench/parts/dataExplorer/electron-browser/nodeActions.contribution.ts

```diff
@@ -16,45 +38,165 @@ MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.IsDatabaseOrServer)
+});
+
+
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'export',
+	order: 7,
+	command: {
+		id: DATA_TIER_WIZARD_COMMAND_ID,
+		title: localize('dacFx', 'Data-tier Application Wizard')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Folder),
+		NodeContextUtils.NodeLabel.isEqualTo('Databases'))
+});
+
+// Profiler
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'profiler',
+	order: 8,
+	command: {
+		id: PROFILER_COMMAND_ID,
+		title: localize('profiler', 'Launch Profiler')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Server))
+});
+
+// Flat File Import
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'import',
+	order: 10,
+	command: {
+		id: IMPORT_COMMAND_ID,
+		title: localize('flatFileImport', 'Import Wizard')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Database))
+});
+
+// Schema Compare
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'export',
+	order: 9,
+	command: {
+		id: SCHEMA_COMPARE_COMMAND_ID,
+		title: localize('schemaCompare', 'Schema Compare')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Database))
+});
+
+// Backup Action
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'connection',
+	order: 4,
+	command: {
+		id: BACKUP_COMMAND_ID,
+		title: localize('backup', 'Backup')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Database))
 });
+
+// Restore Action
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'connection',
+	order: 5,
+	command: {
+		id: RESTORE_COMMAND_ID,
+		title: localize('restore', 'Restore')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Database))
+});
+
+// Generate Scripts Action
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'z-AdminToolExt@1',
+	order: 11,
+	command: {
+		id: GENERATE_SCRIPTS_COMMAND_ID,
+		title: localize('generateScripts', 'Generate Scripts...')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Database),
+		NodeContextUtils.IsWindows)
+});
+
+// Properties Action
+MenuRegistry.appendMenuItem(MenuId.DataExplorerContext, {
+	group: 'z-AdminToolExt@2',
+	order: 12,
+	command: {
+		id: PROPERTIES_COMMAND_ID,
+		title: localize('properties', 'Properties')
+	},
+	when: ContextKeyExpr.and(NodeContextUtils.IsMssqlProvided,
+		NodeContextUtils.NodeType.isEqualTo(NodeType.Server), ContextKeyExpr.not('isCloud'),
+		NodeContextUtils.IsWindows)
+});
```

Comment:
So now every extension that adds new menu items has to also update this core code? That seems very unmaintainable. What if the extension isn't installed? And what about extensions outside our repo?

Instead why not have another dataexplorer extension point and then anyone who registers a menu item can also register it for data explorer. This may not be the most efficient way though - make sure you also chat with Karl and Anthony to get their opinions. 

The ideal solution I think would be to get it so that the nodes in OE and the nodes in data explorer are treated exactly the same - but that's likely to be a bigger design change.

Reply:
Let's talk in person about Data Explorer. I'll go over the Data Explorer and clear your confusion about it.